spatial-lib/src/cmr/spatial/circle.clj

```diff
@@ -0,0 +1,117 @@
+(ns cmr.spatial.circle
+  (:require
+   [cmr.spatial.math :as math :refer :all]
+   [primitive-math]
+   [cmr.spatial.point :as p]
+   [cmr.spatial.derived :as d]
+   [cmr.spatial.geodetic-ring :as gr]
+   [cmr.spatial.validation :as sv]
+   [cmr.spatial.polygon :as poly]
+   [cmr.common.dev.record-pretty-printer :as record-pretty-printer])
+  (:import
+   cmr.spatial.geodetic_ring.GeodeticRing
+   cmr.spatial.polygon.Polygon
+   cmr.spatial.point.Point))
+
+(primitive-math/use-primitive-operators)
+
+;; Circle
+(defrecord Circle
+  [
+   ^Point center
+   ^double radius
+   ])
+
+(record-pretty-printer/enable-record-pretty-printing Circle)
+
+(defn circle
+  "Creates a circle"
+  ([^Point center ^double radius]
+   (->Circle center radius))
+  ([^double lon ^double lat ^double radius]
+   (->Circle (p/point lon lat) radius)))
+
+(def MIN_RADIUS
```

Comment:
where did MIN_RADIUS and MAX_RADIUS come from?

Reply:
We defined them to be reasonable in size. The idea is that circle search is not intended for large areas. This address your suggestion to make the points proportional to radius. I didn't want to make it so fluid at the beginning as I have not done enough tests and have data to show that it would make sense. It is something we can easily add in if we have enough data to see what we should do in the future. Based on my tests in PROD, 16 points is often times slower than 32 points. More than 32 points, the delay will add up. I would like to collect some data before making any optimizations.

Reply:
I will merge this now. I will talk this over with the team again and make necessary changes once we decide on how to pick the number of points of the polygon.